staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -18,6 +18,11 @@ package v1beta1
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+// TODO: This does not generate to protobuf because of the interface{} - fix protobuf
+// generation to support a meta type that can accept any valid JSON.
+
+// +protobuf=false
```

Comment:
Yeah you just need to have a custom marshaller.  runtime.RawExtension is a *very* optimized version of this, but you can look at Time.  Basically you want to turn the struct into json, and then unmarshal a blob of bytes back to your own struct.  Should be easy.